### PR TITLE
[1LP][RFR] Test for checking vm migration after assigning tenant quota

### PR DESCRIPTION
**Pull request by ganeshhubale**
* This test checks whether vm could be able to migrate after assigning tenant quota
    * It creates vm first then assigns tenant quota
    * After assigning tenant quota; it migrates vm and checks migration is done successfully or not

{{ pytest: cfme/tests/infrastructure/test_tenant_quota.py::test_vm_migration_after_assigning_tenant_quota --long-running --use-provider rhv41 }}

### CODE REVIEW

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 pytestmark = [
     test_requirements.quota,
+    pytest.mark.longrunning,
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_tenant_quota.py]: Required: pytest.mark.long_running

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 2]
```diff
@@ -103,6 +104,39 @@ def small_vm(provider, small_template_modscope):
     vm.cleanup_on_provider()
 
 
+@pytest.fixture()
+def new_vm(setup_provider, provider):
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_tenant_quota.py]: Question: Why cannot we use small_vm fixture?

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 3]
```diff
@@ -103,6 +104,39 @@ def small_vm(provider, small_template_modscope):
...
+    """Fixture to provision appliance to the provider being tested if necessary"""
+    vm_name = random_vm_name(context='migrate')
+    try:
+        template_name = provider.data.templates.small_template.name
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_tenant_quota.py]: Required: this check can be added to pytestmark
ex. pytest.mark.provider([VMwareProvider, RHEVMProvider], scope="module", required_fields=[['templates', 'small_template'] selector=ONE_PER_TYPE)

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 4]
```diff
@@ -238,3 +272,49 @@ def test_setting_child_quota_more_than_parent(tenants_setup, parent_quota, child
...
+    """prerequisite: Provider should be added
+
+    steps:
+
+    1. Create VM
+    2. Assign tenant quota
+    3. Migrate VM
+    4. Check whether migration is successfully done
+
+    :param appliance:
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_tenant_quota.py]: You can remove those params from docs as they are fixtures

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 5]
```diff
@@ -238,3 +272,49 @@ def test_setting_child_quota_more_than_parent(tenants_setup, parent_quota, child
...
+    :param new_vm:
+    :param provider:
+    :param tenants_setup:
+    :param parent_quota:
+    :param child_quota:
+    :param flash_text:
+    """
+    view = navigate_to(new_vm, 'Details')
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_tenant_quota.py]: Required: move this to the fixture:
def check_hosts(new_vm, provider):
if  len(provider.hosts) != 1:
view = navigate_to(new_vm, 'Details')
vm_host = view.entities.summary('Relationships').get_text_of('Host')
hosts = [vds.name for vds in provider.hosts if vds.name not in vm_host]
return hosts[0]
else:
pytest.skip("There is only one host in the provider")

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 6]
```diff
@@ -238,3 +272,49 @@ def test_setting_child_quota_more_than_parent(tenants_setup, parent_quota, child
...
+    vm_host = view.entities.summary('Relationships').get_text_of('Host')
+    hosts = [vds.name for vds in provider.hosts if vds.name not in vm_host]
+    if hosts:
+        migrate_to = hosts[0]
+    else:
+        pytest.skip("There is only one host in the provider")
+    new_vm.migrate_vm("email@xyz.com", "first", "last", host=migrate_to)
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_tenant_quota.py]: Required:
use fauxfactory to generate random names and email

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Pull request merged**